onnxruntime/test/framework/memcpy_transformer_test.cc

```diff
@@ -177,6 +178,93 @@ TEST(TransformerTest, MemcpyTransformerTestCudaFirst) {
   ExpectSame(node2, node4, 1);
 }
+TEST(TransformerTest, TestCopyNodeInsertionInitializerInSubgraph) {
+  TensorProto value_tensor;
+  value_tensor.add_dims(1);
+  value_tensor.add_float_data(1.f);
+  value_tensor.set_data_type(ONNX_NAMESPACE::TensorProto_DataType_FLOAT);
+
+  TypeProto tensor_float_type;
+  tensor_float_type.mutable_tensor_type()->set_elem_type(TensorProto_DataType_FLOAT);
+
+  TypeProto tensor_bool_type;
+  tensor_bool_type.mutable_tensor_type()->set_elem_type(TensorProto_DataType_BOOL);
+
+  onnxruntime::NodeArg i1_def("I1", &tensor_bool_type), o1_def("O1", &tensor_float_type), o2_def("O2", &tensor_float_type);
+
+  // main graph
+  std::unordered_map<std::string, int> domain_to_version;
+  domain_to_version[kOnnxDomain] = 7;
+  auto model = std::make_shared<onnxruntime::Model>("test", false, ModelMetaData(), IOnnxRuntimeOpSchemaRegistryList(),
+                                                    domain_to_version);
+  onnxruntime::Graph& graph = model->MainGraph();
+
+  TensorProto parent_constant(value_tensor);
+  parent_constant.set_name("parent_constant");
+  graph.AddInitializedTensor(parent_constant);
+
+  // subgraph
+  std::unordered_map<std::string, int> subgraph_domain_to_version;
+  subgraph_domain_to_version[kOnnxDomain] = 7;
+  auto sub_model = std::make_shared<onnxruntime::Model>("test_subgraph", false, ModelMetaData(), IOnnxRuntimeOpSchemaRegistryList(),
+                                                    subgraph_domain_to_version);
+  onnxruntime::Graph& subgraph = sub_model->MainGraph();
+
+  TensorProto local_constant(value_tensor);
+  local_constant.set_name("local_constant");
+  subgraph.AddInitializedTensor(local_constant);
+
+  subgraph.AddOuterScopeNodeArg("parent_constant");
+  subgraph.AddNode("node1", "Add", "operator1", ArgMap{&subgraph.GetOrCreateNodeArg("local_constant", &tensor_float_type), 
+                                                                     &graph.GetOrCreateNodeArg("parent_constant", &tensor_float_type)}, 
+                                                              ArgMap{&o1_def});
+
+  subgraph.AddNode("node2", "Add", "operator2", ArgMap{&subgraph.GetOrCreateNodeArg("local_constant", &tensor_float_type), 
+                                                                     &graph.GetOrCreateNodeArg("parent_constant", &tensor_float_type)},
+                                                              ArgMap{&o2_def});
+
+  auto status = subgraph.Resolve();
+  ASSERT_TRUE(status.IsOK()) << status.ErrorMessage();
+
+ // main graph continued
+  auto& if_node = graph.AddNode("node3", "If", "cpu operator2", ArgMap{&i1_def}, ArgMap{&o1_def, &o2_def});
+  if_node.AddAttribute("then_branch", {subgraph.ToGraphProto()});
+  if_node.AddAttribute("else_branch", {subgraph.ToGraphProto()});
+
+  onnxruntime::Graph* subgraph_1 = if_node.GetMutableGraphAttribute("then_branch");
+  for (auto& node : subgraph_1->Nodes()) {
+    if (node.Name() == "node2") {
```

Comment:
Some comments here saying what you're doing and why would help someone new understand this test case.